jep/0000/README.adoc

```diff
@@ -0,0 +1,329 @@
+
+| Type
+| Standards
+
+| Created
+| 2018-07-24
+
+| BDFL-Delegate
+| TBD
+
+//
+//
+// Uncomment if there is an associated placeholder JIRA issue.
+//| JIRA
+//| :bulb: https://issues.jenkins-ci.org/browse/JENKINS-nnnnn[JENKINS-nnnnn] :bulb:
+//
+//
+// Uncomment if discussion will occur in forum other than jenkinsci-dev@ mailing list.
+//| Discussions-To
+//| :bulb: Link to where discussion and final status announcement will occur :bulb:
+//
+//
+// Uncomment if this JEP depends on one or more other JEPs.
+| Requires
+| link:/jep/jep-207[JEP-207], link:/jep/jep-210[JEP-210]
+//
+//
+// Uncomment and fill if this JEP is rendered obsolete by a later JEP
+//| Superseded-By
+//| :bulb: JEP-NUMBER :bulb:
+//
+//
+// Uncomment when this JEP status is set to Accepted, Rejected or Withdrawn.
+//| Resolution
+//| :bulb: Link to relevant post in the jenkinsci-dev@ mailing list archives :bulb:
+
+|===
+
+== Abstract
+
+On large-scale Jenkins instances master Disk and Network I/O become bottlenecks in particular cases.
+Build logging and Task reporting are one for the most intensive I/O consumers,
+hence it would be great to somehow redirect them to an external system.
+This is a continuation of the original story we had back in 2016
+(see the public design document
+link:TODO[here]).
+JEP-TODO defines requirements to the external logging system and Core APIs.
+
+The External Logging API plugin is a separate plugin
+which offers shared functionality to External Logging implementations.
+
+* Configuration UI
+* Extension Points for External Logging implementations
+* Base classes for external logging,
+which simplify implementations
+* Other shared utility classes
+
+== Specification
+
+=== Event class
+
+Event is an atomic entry of the external logging.
+In the data model it includes the following fields:
+
+* `id` (long) - ID of the message.
+  It must be unique within a single `LoggingMethod` writer
+* `message`(string) - Contains the message body.
+  It may be a single-line or a multi-line message.
+  The message may also contain annotations.
+* `timestamp` (long) - Timestamp of the message,
+  measured in milliseconds, between
+  the current time and midnight, January 1, 1970 UTC
+* `data` (`Map<String, Serializable>`) - additional metadata.
+  This metadata will be used to categorize event entries.
+
+Some metadata is mandatory.
+Requited fields are defined below.
+
+==== Event metadata
+
+The following metadata entries are required to be present
+for all event entries:
+
+* `jenkinsId` - Instance ID of the Jenkins instance
+* `loggableType`  - Type of the `Loggable` entry, e.g. `run`
+* `loggableId` - Identifier of the `Loggable` entry
+
+==== Run metadata
+
+The following metadata entries are required to be present
+for all `hudson.model.Run` event entries:
+
+* All event metadata
+* `jobId` - ID of the Job, should be provided by Unique ID Plugin
+
+==== Pipeline metadata
+
+* All Run metadata
+* `stepId` - Identifier of the step ("node")
```

Comment:
I think you probably mean flowNodeId, since that's the actual object that is mapped to where a log originates from (1 or more flownodes per step - an important key point).